bundler/lib/bundler/settings.rb

```diff
@@ -419,7 +419,15 @@ def printable_value(value, key)
       elsif is_credential(key)
         "[REDACTED]"
       elsif is_userinfo(converted)
-        converted.gsub(/:.*$/, ":[REDACTED]")
+        username, pass = converted.split(":")
```

Comment:
After seeing the updated code, I'm actually not sure the new code is more clear. With the previous version, it was kind of clear that `x-oauth-basic` at the place of the password was an special case where the username should be treated as the redactable part. Even Github docs talk about

> You can simply use an OAuth token for the username and either a blank password or the string x-oauth-basic when cloning a repository.

Sorry @mlarraz for the back and forth, don't update this yet, if @intuxicated still feels `key:value` is better, we can stick to this version, it's no big deal.

Reply:
Yeah, you are right @deivid-rodriguez. I look into the ruby implementation of [URI::Generic#split_userinfo](https://github.com/ruby/ruby/blob/f51a6ace06cecd193bf00747a63809849e254050/lib/uri/generic.rb#L540-L548) and it's the same thing there, `username, password`. I think the only improvement here would be providing the integer `2` as the second argument to prevent unexpected behavior.

```ruby
irb(main)>  URI('http://user:password@ecample.com').userinfo.split(':')
=> ["user", "password"] # ok
irb(main)>  URI('http://user:password@ecample.com').userinfo.split(':', 2)
=> ["user", "password"] # ok
irb(main)>  URI('http://user:pass:word@ecample.com').userinfo.split(':')
=> ["user", "pass", "word"] # bad
irb(main)>  URI('http://user:pass:word@ecample.com').userinfo.split(':', 2)
=> ["user", "pass:word"] # prefect - cover both scenario

```

Reply:
Definitely :+1:

Reply:
Sorry, @mlarraz. If you could drop the `8e88d08` from your PR, it would be awesome. Also, add the second argument with the value of `2` to make sure we are not breaking things.
Also, If you don't have the time, you can leave it to me, I'll fix it. Thanks 🥇